coordinator/src/tributary/scanner.rs

```diff
@@ -114,12 +117,38 @@ pub(crate) async fn handle_new_blocks<
   processors: &Pro,
   publish_serai_tx: PST,
   spec: &TributarySpec,
-  tributary: &TributaryReader<D, Transaction>,
+  tributary: &Arc<Tributary<D, Transaction, P>>,
```

Comment:
You shouldn't need to take `&Arc`.  `&Arc::new(x)` should deref to `&x`.

I'm unsure the impact of this (taking the Tributary) on latency. It shouldn't be atrocious with the recent localization I did, yet even if it's perfectly fine (which I'd hope), it shouldn't be necessary. We should add a method to the reader to check if an on-chain provided transaction was provided into the tributary ourselves yet.

IIRC, `Provided` are perfectly ordered within some queue, yet aren't guaranteed to have distinct hashes. Accordingly, on local provision, you'd save `Provided queue i hash`. Then, on-chain when you see a Provided, you'd check `Provided queue i hash`.

There is a pseudo-unsafety here, where `Provided queue i` is insufficient as what other validators came to may be distinct from what we came to. Under this, we'd likely enter a wait loop. The question is if the tributary should return an error in such a case. It likely should.